.github/workflows/github.yml

```diff
@@ -385,6 +435,42 @@ jobs:
             name: ${{ env.VCMI_PACKAGE_FILE_NAME }}
             path: |
               ./release.tar.gz
+  
+  extract-version:
+    name: Extract Version
+    runs-on: ubuntu-24.04
+    outputs:
+      version_timestamp: ${{ steps.extract.outputs.version_timestamp }}
+      short_version: ${{ steps.extract.outputs.short_version }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Extract version info
+        id: extract
+        shell: pwsh
+        run: |
+          $filePath = "$env:GITHUB_WORKSPACE/cmake_modules/VersionDefinition.cmake"
+          if (!(Test-Path $filePath)) {
+            Write-Host "File not found: $filePath"
+            exit 1
+          }
+
+          $content = Get-Content $filePath -Raw
+
+          $majorMatch = [regex]::Match($content, 'set\(VCMI_VERSION_MAJOR\s+(\d+)\)')
+          $minorMatch = [regex]::Match($content, 'set\(VCMI_VERSION_MINOR\s+(\d+)\)')
+          $patchMatch = [regex]::Match($content, 'set\(VCMI_VERSION_PATCH\s+(\d+)\)')
+
+          if (!($majorMatch.Success -and $minorMatch.Success -and $patchMatch.Success)) {
+            Write-Host "Failed to extract version numbers"
+            exit 1
+          }
+
+          $shortVersion = "$($majorMatch.Groups[1].Value).$($minorMatch.Groups[1].Value).$($patchMatch.Groups[1].Value)"
+          $timestamp = Get-Date -Format "yyyyMMddHHmmss"
```

Comment:
simply using `grep` in bash seems like an easier alternative. Besides, I don't think we have PS speakers in the project, I wouldn't use it unless it greatly simplifies things.